File: Src/Data/Entities/Track.cs
namespace MusicPlayerBackend.Data.Entities;

public enum TrackVisibility
{
    Hidden = 0,
    Visible = 1
}

public sealed record Track : EntityBase
{
    public string? CoverUri { get; set; }
    public string? TrackUri { get; set; }
    public TrackVisibility Visibility { get; set; }

    public string Name { get; set; } = null!;
    public string Author { get; set; } = null!;

    public Guid OwnerUserId { get; set; }
    public User OwnerUser { get; set; } = null!;
}

File: Src/Persistence/Entities/Track.fs
namespace MusicPlayerBackend.Persistence.Entities

open System

type TrackId = Guid

type TrackVisibility =
    | Hidden = 0
    | Visible = 1

type Track = {
    mutable Id: TrackId
    mutable OwnerUserId: UserId

    mutable CoverUri: string option
    mutable TrackUri: string option
    mutable Visibility: TrackVisibility

    mutable Name: string
    mutable Author: string

    mutable CreatedAt: DateTimeOffset
    mutable UpdatedAt: DateTimeOffset voption
}
